packages/agent/src/events.ts

```diff
@@ -35,13 +33,11 @@ export default class Events<T> implements AgentEventBase<T> {
   }
   public on (
     eventName: EventType,
-    handler: (event: T) => void,
-    type: InteractType
+    handler: (event: T) => void
   ): void {
-    if (typeof handler !== 'function' || !(type === 'a' || type === 'l')) {
+    if (typeof handler !== 'function') {
```

Comment:
Sorry, this block has been dead...because handler is typing as `event: T => void`.
Please delete this block.